app/dao/notifications_dao.py

```diff
@@ -267,6 +267,31 @@ def get_notifications_for_job(
     return pagination
 
 
+def get_recent_notifications_for_job(
+    service_id, job_id, filter_dict=None, page=1, page_size=None
+):
+    print(f"FILTER_DICT AT DAO LEVEL {filter_dict}")
+    if page_size is None:
+        page_size = current_app.config["PAGE_SIZE"]
+
+    stmt = select(Notification).where(
+        Notification.service_id == service_id,
+        Notification.job_id == job_id,
+    )
+
+    stmt = _filter_query(stmt, filter_dict)
+    stmt = stmt.order_by(desc(Notification.job_row_number))
+    print(f"STMT {stmt}")
```

Comment:
Another `print` statement - same question about whether it should be converted to logging or not.

```suggestion
```